### ISSUE TITLE
Intermediate error catch event can be created via arrow keys element movement

### ISSUE THREAD

**Issue by barmac** [bpmn-io/bpmn-js#1803]
__Describe the Bug__

Given I have an error boundary event, when I press arrow keys, I can create an intermediate error catch event which is not allowed in BPMN.

https://user-images.githubusercontent.com/28307541/212301657-da2554cb-9691-4f63-9b1c-efc4f61764d9.mov

__Steps to Reproduce__

1. Create an error boundary event and select it
2. Press arrow keys to detach it
3. ???

__Expected Behavior__

No such stuff is allowed. Either error event definition should be dropped, or the action should be ignored (element should not move).

__Environment__

 - Browser: [e.g. IE 11, Chrome 69]
 - OS: [e.g. Windows 7]
 - Library version: [e.g. 2.0.0]

**Comment by marstamm** [comment]
## Findings 

- The element gets detached as we don't pass hints or parents to `moveElements` in the [`KeyboardMoveSelection` module](https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/keyboard-move-selection/KeyboardMoveSelection.js#L130-L132).
- It currently offers no way to hook into the move lifecycle like we have in (mouse) `Move` that other features can hook into (e.g. [AttachSupport](https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/attach-support/AttachSupport.js#L48)).

To fix this, we need to
- Refactor `KeyboardMoveSelection` to emit events and allow changes before move and rules check
- Refactor `AttachSupport` to hook into these events
- (optional) update [`BpmnRules` attach support](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/rules/BpmnRules.js#L624-L626) to check if the element is out of bounds. It currently only checks if the boundary completely intersects with the host

**Comment by philippfromme** [comment]
We could allow the detachment and simply remove the event definition when the element is replaced. We don't allow detaching through mouse interaction but we could.

![electron_AUC9T8zXaD](https://github.com/user-attachments/assets/bc98e843-21ab-4edb-b2ce-36e8f19be339)

**Comment by AlekseyManetov** [comment]
I'd propose the following fix:

We could disable keyboard movement for boundary events if they are moved without their parent. From my perspective, it makes no sense to move them using the keyboard. Otherwise, accidental detachment of the event from the parent due to a mistaken arrow key press could be confusing and break the diagram without the user noticing.

We can do this by improving our BPMN `elements.move` rule, so it will affect only BPMN diagrams and leave element moving support for more generic `diagram-js` cases.

https://github.com/user-attachments/assets/0ee5522d-0814-4951-aa0e-ccefe295661a

**Comment by philippfromme** [comment]
So to generalize: Disallow keyboard movement of any attacher without its host. I think this a good solution. You could argue that we don't allow the users to do this anymore:

![Image](https://github.com/user-attachments/assets/44674d7e-5649-4aaf-8e8d-a762751c49d2)

But I'm not sure if that is even a use case. I think keyboard movement is a cool feature but I've never used it. 🙈

**Comment by barmac** [comment]
> You could argue that we don't allow the users to do this anymore:

Right, but this is actually a bug since once you move the element with the keyboard, it's not attached anymore. It only looks attached, but the shape type has already changed.

**Comment by barmac** [comment]
So either we keep the attachment after move, or we disallow movement at all. But the feature is unusable at the current state.

**Comment by philippfromme** [comment]
Okay, simply disabling it should be fine.

**Comment by nikku** [comment]
Fixed in [`v18.12.0`](https://github.com/bpmn-io/bpmn-js/releases/tag/v18.12.0).
